From 343a896d00f3d5e73f7192e8cd1fd4690ea25739 Mon Sep 17 00:00:00 2001
From: Takuro Ashie <ashie@clear-code.com>
Date: Thu, 11 May 2017 14:13:54 +0900
Subject: [PATCH 2/4] OmxDecoderModule: Fix a bug which crashes about:support
 page

In the current version TaskQueue should be recieved via
CreateDecoderParams instead of using AbstractThread::GetCurrent().

Signed-off-by: Takuro Ashie <ashie@clear-code.com>
---
 dom/media/platforms/omx/OmxDataDecoder.cpp   | 6 +++++-
 dom/media/platforms/omx/OmxDataDecoder.h     | 2 ++
 dom/media/platforms/omx/OmxDecoderModule.cpp | 4 ++++
 3 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/dom/media/platforms/omx/OmxDataDecoder.cpp b/dom/media/platforms/omx/OmxDataDecoder.cpp
index db3a32b7e91d..719ae5dd7864 100644
--- a/dom/media/platforms/omx/OmxDataDecoder.cpp
+++ b/dom/media/platforms/omx/OmxDataDecoder.cpp
@@ -100,8 +100,12 @@ protected:
 };
 
 OmxDataDecoder::OmxDataDecoder(const TrackInfo& aTrackInfo,
+                               TaskQueue* aTaskQueue,
+                               MediaDataDecoderCallback* aCallback,
                                layers::ImageContainer* aImageContainer)
-  : mOmxTaskQueue(CreateMediaDecodeTaskQueue("OmxDataDecoder::mOmxTaskQueue"))
+  : mMonitor("OmxDataDecoder")
+  , mOmxTaskQueue(CreateMediaDecodeTaskQueue())
+  , mReaderTaskQueue(aTaskQueue)
   , mImageContainer(aImageContainer)
   , mWatchManager(this, mOmxTaskQueue)
   , mOmxState(OMX_STATETYPE::OMX_StateInvalid, "OmxDataDecoder::mOmxState")
diff --git a/dom/media/platforms/omx/OmxDataDecoder.h b/dom/media/platforms/omx/OmxDataDecoder.h
index d6199b5b29ba..f3e65e8d3645 100644
--- a/dom/media/platforms/omx/OmxDataDecoder.h
+++ b/dom/media/platforms/omx/OmxDataDecoder.h
@@ -67,6 +67,8 @@ protected:
 
 public:
   OmxDataDecoder(const TrackInfo& aTrackInfo,
+                 TaskQueue* aTaskQueue,
+                 MediaDataDecoderCallback* aCallback,
                  layers::ImageContainer* aImageContainer);
 
   RefPtr<InitPromise> Init() override;
diff --git a/dom/media/platforms/omx/OmxDecoderModule.cpp b/dom/media/platforms/omx/OmxDecoderModule.cpp
index 69e94b6c63c0..8b014534082a 100644
--- a/dom/media/platforms/omx/OmxDecoderModule.cpp
+++ b/dom/media/platforms/omx/OmxDecoderModule.cpp
@@ -39,6 +39,8 @@ already_AddRefed<MediaDataDecoder>
 OmxDecoderModule::CreateVideoDecoder(const CreateDecoderParams& aParams)
 {
   RefPtr<OmxDataDecoder> decoder = new OmxDataDecoder(aParams.mConfig,
+                                                      aParams.mTaskQueue,
+                                                      aParams.mCallback,
                                                       aParams.mImageContainer);
   return decoder.forget();
 }
@@ -47,6 +49,8 @@ already_AddRefed<MediaDataDecoder>
 OmxDecoderModule::CreateAudioDecoder(const CreateDecoderParams& aParams)
 {
   RefPtr<OmxDataDecoder> decoder = new OmxDataDecoder(aParams.mConfig,
+                                                      aParams.mTaskQueue,
+                                                      aParams.mCallback,
                                                       nullptr);
   return decoder.forget();
 }
-- 
2.14.1

